index.channel.ts

```diff
@@ -0,0 +1,759 @@
+      // TODO: handle non compact urls with link unfurling: https://api.slack.com/reference/messaging/link-unfurling#event_deliveries
+      if (btn.type === ButtonType.web_url) {
+        return {
+          type: 'button',
+          text: {
+            type: 'plain_text',
+            text: btn.title,
+            emoji: true,
+          },
+          value: 'url',
+          url: btn.url,
+        };
+      } else {
+        return {
+          type: 'button',
+          text: {
+            type: 'plain_text',
+            text: btn.title,
+            emoji: true,
+          },
+          value: btn.payload,
+        };
+      }
+    });
+    return {
+      blocks: [textSection, { type: 'actions', elements }],
+    };
+  }
+
+  //TODO: get usersList
+
+  //This method will return undefined if the quick replies are not present
+
+  /**
+   * Uploads the attachment file to Slack and formats the quick replies if present
+   *
+   * @param message - An attachement + quick replies to be sent to the end user
+   * @param channel - The slack channels to send the message to, separated by commas
+   * @param options - Might contain additional settings
+   * @returns
+   */
+  async _attachmentFormat(
+    message: StdOutgoingAttachmentMessage<WithUrl<Attachment>>,
+    channel: string,
+    options?: any,
+  ): Promise<Slack.OutgoingMessage> {
+    const fileUploader = new SlackFileUploader(
+      this.api,
+      message.attachment,
+      channel,
+      this.attachmentService,
+    );
+    await fileUploader.upload();
+
+    if (message.quickReplies?.length > 0)
+      return this._quickRepliesFormat({
+        text: '',
+        quickReplies: message.quickReplies,
+      });
+    return undefined;
+  }
+
+  /**
+   * Formats a collection of elements to be sent to Slack in carousel/list format
+   *
+   * @param data - A list of data items to be sent to the end user
+   * @param options - Might contain additional settings
+   * @returns - A Blocks array of Slack elements
+   */
+  _formatElements(data: any[], options: any, ...args: any): any[] {
+    const fields = options.content.fields;
+    const buttons = options.content.buttons;
+    //To build a list :
+    const blocks: Slack.KnownBlock[] = [{ type: 'divider' }];
+    data.forEach((item) => {
+      const text = item[fields.subtitle]
+        ? '*' + item[fields.title] + '*\n' + item[fields.subtitle]
+        : '*' + item[fields.title] + '*';
+      //Block containing the title and subtitle and image
+      const main_block: Slack.SectionBlock = {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text,
+        },
+      };
+
+      if (item[fields.image_url])
+        main_block.accessory = {
+          type: 'image',
+          image_url: item[fields.image_url].payload.url,
+          alt_text: item[fields.title],
+        };
+      blocks.push(main_block);
+      //Array of elements : Buttons
+      const elements = [];
```

Comment:
```suggestion
      const elements = [];
```
Can we define the type for elements